server/task-cassandra/src/main/scala/org/apache/james/task/eventsourcing/cassandra/CassandraTaskExecutionDetailsProjectionDAO.scala

```diff
@@ -0,0 +1,84 @@
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+package org.apache.james.task.eventsourcing.cassandra
+
+import java.util.Optional
+
+import javax.inject.Inject
+
+import org.apache.james.backends.cassandra.init.{CassandraTypesProvider, CassandraZonedDateTimeModule}
+import org.apache.james.backends.cassandra.utils.CassandraAsyncExecutor
+import org.apache.james.task.eventsourcing.cassandra.CassandraTaskExecutionDetailsProjectionTable._
+import org.apache.james.task.{TaskExecutionDetails, TaskId, TaskManager}
+
+import com.datastax.driver.core.querybuilder.QueryBuilder
+import com.datastax.driver.core.querybuilder.QueryBuilder.{bindMarker, insertInto, select}
+import com.datastax.driver.core.{Row, Session}
+import reactor.core.publisher.{Flux, Mono}
+
+@Inject
+class CassandraTaskExecutionDetailsProjectionDAO(val session: Session, val typesProvider: CassandraTypesProvider) {
+
+  private val cassandraAsyncExecutor = new CassandraAsyncExecutor(session)
+  private val dateType = typesProvider.getDefinedUserType(CassandraZonedDateTimeModule.ZONED_DATE_TIME)
+
+  private val insertStatement = session.prepare(insertInto(TABLE_NAME)
+    .value(TASK_ID, bindMarker(TASK_ID))
+    .value(TYPE, bindMarker(TYPE))
+    .value(STATUS, bindMarker(STATUS))
+    .value(SUBMITTED_DATE, bindMarker(SUBMITTED_DATE))
+    .value(STARTED_DATE, bindMarker(STARTED_DATE))
+    .value(COMPLETED_DATE, bindMarker(COMPLETED_DATE))
+    .value(CANCELED_DATE, bindMarker(CANCELED_DATE))
+    .value(FAILED_DATE, bindMarker(FAILED_DATE)))
+
+  private val selectStatement = session.prepare(select().from(TABLE_NAME)
+    .where(QueryBuilder.eq(TASK_ID, bindMarker(TASK_ID))))
+
+  private val listStatement = session.prepare(select().from(TABLE_NAME))
+
+  def saveDetails(details : TaskExecutionDetails): Mono[Void] = cassandraAsyncExecutor.executeVoid(
+    insertStatement.bind
+      .setUUID(TASK_ID, details.getTaskId.getValue)
+      .setString(TYPE, details.getType)
+      .setString(STATUS, details.getStatus.getValue)
+      .setUDTValue(SUBMITTED_DATE, CassandraZonedDateTimeModule.toUDT(dateType, details.getSubmitDate).orElse(null))
+      .setUDTValue(STARTED_DATE, CassandraZonedDateTimeModule.toUDT(dateType, details.getStartedDate).orElse(null))
+      .setUDTValue(COMPLETED_DATE, CassandraZonedDateTimeModule.toUDT(dateType, details.getCompletedDate).orElse(null))
+      .setUDTValue(CANCELED_DATE, CassandraZonedDateTimeModule.toUDT(dateType, details.getCanceledDate).orElse(null))
+      .setUDTValue(FAILED_DATE, CassandraZonedDateTimeModule.toUDT(dateType, details.getStartedDate).orElse(null)))
+
+  def readDetails(taskId: TaskId): Mono[TaskExecutionDetails] = cassandraAsyncExecutor.executeSingleRow(
+    selectStatement.bind().setUUID(TASK_ID, taskId.getValue))
+      .map(readRow)
```

Comment:
the indent shouldn't be reversed between this line and the previous one?

Reply:
Euuuuhhh no?

Reply:
`.map` is in a upper level, same as `executeSingleRow`. 
`selectStatement` is inside `executeSingleRow`, so why is `map` indented more and not the contrary?

Reply:
@blackheaven I would like to have the reason why you think my reasoning is not right though, instead of a thumbs up on a no :P